great_expectations/rule_based_profiler/domain_builder/domain.py

```diff
@@ -1,39 +1,98 @@
+import json
 from typing import Any, Dict, Optional, Union
 
 from great_expectations.core import IDDict
 from great_expectations.core.util import convert_to_json_serializable
+from great_expectations.execution_engine.execution_engine import MetricDomainTypes
+from great_expectations.rule_based_profiler.domain_builder.inferred_semantic_domain_type import (
+    SemanticDomainTypes,
+)
 from great_expectations.types.base import SerializableDotDict
+from great_expectations.util import filter_properties_dict
 
 
-# TODO: <Alex>ALEX -- multiple inheritance is to be used with caution (potentially fix to use single inheritance)</Alex>
-class Domain(IDDict, SerializableDotDict):
+class DomainKwargs(SerializableDotDict):
+    def to_json_dict(self) -> dict:
+        return convert_to_json_serializable(data=dict(self))
+
+
+class Domain(SerializableDotDict):
     # Adding an explicit constructor to highlight the specific properties that will be used.
-    def __init__(self, domain_kwargs: Optional[Dict[str, Any]] = None):
+    def __init__(
+        self,
+        domain_type: Union[str, MetricDomainTypes, SemanticDomainTypes],
+        domain_kwargs: Optional[Union[Dict[str, Any], DomainKwargs]] = None,
+    ):
+        if not domain_type:
+            raise ValueError("Cannot instantiate Domain (domain_type is required).")
```

Comment:
Tiny nitpick: Since `domain_type` is not an optional argument, python will throw a `TypeError` for us so these lines wouldn't run.

Reply:
@anthonyburdi Great catch!  It used to be optional, but I forgot to remove the check.  Thanks! (in next commit).